service-spec/service-spec-commons/src/main/java/io/hyscale/servicespec/commons/util/ImageUtil.java

```diff
@@ -28,29 +28,64 @@
 public class ImageUtil {
 
     private static final String DELIMITER = "/";
-    
+
     /**
      * Get complete image name from service spec
-     * @param serviceSpec
+     * @param serviceSpec 
      * @return image complete name
      * registry url + "/" + imageName:tag
      * 1. registry url is null - imageName:tag
      * 2. registry url and tag not present - imageName
      * @throws HyscaleException
      */
     public static String getImage(ServiceSpec serviceSpec) throws HyscaleException {
+        if(serviceSpec == null){
+            return null;
+        }
+        StringBuilder sb = new StringBuilder();
+        sb.append(getImageWithoutTag(serviceSpec));
         String tag = serviceSpec.get(HyscaleSpecFields.getPath(HyscaleSpecFields.image, HyscaleSpecFields.tag), String.class);
```

Comment:
Service spec.get(servicespec,ImageDeserializer.class) should work right instead of always parsing the service spec.... ?

Reply:
Fix the ImageDeserialiazer first to serialize for both DockerBUildImage & BuildSpec .